apps/module-pool/contracts/Pool.sol

```diff
@@ -7,41 +7,57 @@ pragma solidity 0.4.24;
-    string private constant ERROR_TOKEN_ALREADY_EXISTS = "POOL_TOKEN_ALREADY_EXISTS";
-    string private constant ERROR_TOKEN_DOES_NOT_EXIST = "POOL_TOKEN_DOES_NOT_EXIST";
-    string private constant ERROR_TARGET_IS_GUARDED = "POOL_TARGET_IS_GUARDED";
+    string private constant ERROR_TOKEN_ALREADY_PROTECTED = "POOL_TOKEN_ALREADY_PROTECTED";
+    string private constant ERROR_TOKEN_NOT_PROTECTED = "POOL_TOKEN_NOT_PROTECTED";
+    string private constant ERROR_TARGET_PROTECTED = "POOL_TARGET_PROTECTED";
+    string private constant ERROR_PROTECTED_TOKENS_MODIFIED = "POOL_PROTECTED_TOKENS_MODIFIED";
     string private constant ERROR_BALANCE_NOT_CONSTANT = "POOL_BALANCE_NOT_CONSTANT";
 
-    mapping (uint256 => address) public collateralTokens;
-    uint256 public collateralTokensLength;
+    uint256 public constant PROTECTED_TOKENS_CAP = 10;
+    address[] public protectedTokens;
 
     event SafeExecute(address indexed sender, address indexed target, bytes data);
-    event AddCollateralToken(address indexed token);
-    event RemoveCollateralToken(address indexed token);
+    event AddProtectedToken(address indexed token);
+    event RemoveProtectedToken(address indexed token);
 
     /***** external functions *****/
 
     /**
-    * @notice Safe execute '`@radspec(_target, _data)`' on `_target`
-    * @param _target Address where the action is being executed
-    * @param _data Calldata for the action
-    * @return Exits call frame forwarding the return data of the executed call (either error or success data)
+     * @notice Safe execute '`@radspec(_target, _data)`' on `_target`
+     * @param _target Address where the action is to be executed
+     * @param _data Calldata for the action to be executed
+     * @return Exits call frame forwarding the return data of the executed call [either error or success data]
     */
     function safeExecute(address _target, bytes _data) external auth(SAFE_EXECUTE_ROLE) {
-        uint256[] memory balances = new uint256[](collateralTokensLength);
+        address[] memory _protectedTokens = new address[](protectedTokens.length);
+        uint256[] memory balances = new uint256[](protectedTokens.length);
         bytes32 size;
         bytes32 ptr;
 
-        for (uint256 i = 0; i < collateralTokensLength; i++) {
-            address token = collateralTokens[i + 1];
-            // we don't care if target is ETH [0x00...] as it can't be spent anyhow and users may will to burn non-guarded tokens for some reason ...
-            require(token == ETH || token != _target, ERROR_TARGET_IS_GUARDED);
+        for (uint256 i = 0; i < protectedTokens.length; i++) {
+            address token = protectedTokens[i];
+            // we don't care if target is ETH [0x00...0] as it can't be spent anyhow [though you can't invoke anything at 0x00...0]
+            require(_target != token || token == ETH, ERROR_TARGET_PROTECTED);
```

Comment:
@osarrouy it could make sense to require that `_target` cannot be the `controller` to prevent that the `SAFE_EXECUTE_ROLE` can call out to the methods of `controller` (buy,sell) that are likely to be set to allow `ANY_ENTITY`.